contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -2,75 +2,227 @@
 pragma solidity ^0.8.19;
 
 import {IForwarder} from "./interfaces/IForwarder.sol";
+import {IReceiver} from "./interfaces/IReceiver.sol";
 import {ConfirmedOwner} from "../shared/access/ConfirmedOwner.sol";
 import {TypeAndVersionInterface} from "../interfaces/TypeAndVersionInterface.sol";
-import {Utils} from "./libraries/Utils.sol";
 
 // solhint-disable gas-custom-errors, no-unused-vars
```

Comment:
@DeividasK add comment:
This is an entry point for Write Chain Target capability. This allows us to determine if reports have been processed (successfully or not) in a generic way so that products don't need to validate delivery on an ad-hoc basis.

For example, users create a Workflow Spec that wants to deliver a custom report to Ethereum Mainnet. We need to be able to report back to the user on successful (or unsuccessful) delivery. This contract uses execution ID and recipient address to store the results of attempted deliveries. This way, other nodes can also verify if something was delivered on-chain.

The nodes supporting the Target capability would interact with the Forwarder contract.